include/seqan3/io/stream/parse_condition_detail.hpp

```diff
@@ -0,0 +1,490 @@
+ * \brief Provides parse conditions for tokenization.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <cctype>
+#include <stdexcept>
+#include <string>
+
+#include <seqan3/std/concepts>
+
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/core/detail/reflection.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/io/exception.hpp>
+#include <seqan3/range/container/constexpr_string.hpp>
+
+namespace seqan3::detail
+{
+
+// ----------------------------------------------------------------------------
+// constexpr_pseudo_bitset (could be replaced with constexpr std::bitset or github.com/ClaasBontus/bitset2)
+// ----------------------------------------------------------------------------
+
+/*!\brief A data structure that implements a subset of std::bitset as constexpr.
+ * \ingroup stream
+ * \tparam N                The number of bits.
+ */
+template <size_t N>
+class constexpr_pseudo_bitset : public std::array<bool, N>
+{
+private:
+    //!\brief The base type.
+    using base_t = std::array<bool, N>;
+public:
+    //!\brief Inherit constructors.
+    using base_t::base_t;
+
+    //!\brief Return a new bitset that is a logical disjunction of the two given ones.
+    constexpr constexpr_pseudo_bitset operator|(constexpr_pseudo_bitset rhs) const noexcept
+    {
+        for (size_t i = 0; i < N; ++i)
+            rhs[i] = rhs[i] || base_t::operator[](i);
+
+        return std::move(rhs);
+    }
+    //!\brief Return a new bitset with all bits flipped.
+    constexpr constexpr_pseudo_bitset operator~() const noexcept
+    {
+        constexpr_pseudo_bitset ret{};
+        for (size_t i = 0; i < N; ++i)
+            ret[i] = !base_t::operator[](i);
+
+        return ret;
+    }
+};
+
+// ----------------------------------------------------------------------------
+// condition_message_v
+// ----------------------------------------------------------------------------
+
+/*!\brief Defines a compound seqan3::constexpr_string consisting of all given conditions separated by the
+ *        operator-name `op`.
+ * \ingroup stream
+ * \tparam op               non-type template parameter specifying the separator character, e.g. '|'.
+ * \tparam condition_head_t The first condition type in the message. Ensures that there is at least one type.
+ * \tparam condition_ts     Remaining list of conditions separated by `op`.
+ * \relates seqan3::detail::parse_condition
+ */
+template <char op, typename condition_head_t, typename ...condition_ts>
+inline constexpr_string constexpr condition_message_v
+{
+    constexpr_string{"("} +
+    (condition_head_t::msg + ... +
+        (constexpr_string{" "} + constexpr_string{{op, op, '\0'}} + constexpr_string{" "} + condition_ts::msg)) +
+    constexpr_string{")"}
+};
+
+// ----------------------------------------------------------------------------
+// parse_condition_concept
+// ----------------------------------------------------------------------------
+
+//!\cond
```

Comment:
shouldn't the base class be at least documented in DEV?

Reply:
it's just a forward decleration!